src/Servers/Kestrel/test/InMemory.FunctionalTests/Http3/Http3TestBase.cs

```diff
@@ -221,6 +288,46 @@ protected static async Task FlushAsync(PipeWriter writableBuffer)
             {
                 await writableBuffer.FlushAsync().AsTask().DefaultTimeout();
             }
+
+            internal async Task<Http3FrameWithPayload> ReceiveFrameAsync(uint maxFrameSize = 16 * 1024)
+            {
+                var frame = new Http3FrameWithPayload();
+
+                while (true)
+                {
+                    var result = await _pair.Application.Input.ReadAsync().AsTask().DefaultTimeout();
+                    var buffer = result.Buffer;
+                    var consumed = buffer.Start;
+                    var examined = buffer.Start;
+                    var copyBuffer = buffer;
+
+                    try
+                    {
+                        Assert.True(buffer.Length > 0);
+
+                        if (Http3FrameReader.TryReadFrame(ref buffer, frame, maxFrameSize, out var framePayload))
```

Comment:
I looked in `Http3FrameReader` to see what it did with maxFrameSize, and it looks unused. Is there a maximum frame size in HTTP/3?

The reason I started looking into this is to see how header continuation frames were handled, but they don't seem to be a thing in HTTP/3.

Reply:
I don't think there is a maximum frame size in HTTP/3, but a peer may enforce one for headers via a HTTP/3 setting: https://quicwg.org/base-drafts/draft-ietf-quic-http.html#section-4.1.1.3.

So I think maxFrameSize should somehow only apply to headers.